### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by westerbarendregt**
Pull Request of Guilds and guildrequests. notes on these models and modules would be appreciated
Wars are in progress and this is not the latests version of the wars, notes won't be necessary here.

### CODE REVIEW

File: vivid-server/src/modules/guildrequest/guildrequest.service.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    @InjectRepository(GuildRequestEntity)
+    private guildRequestRepository: Repository<GuildRequestEntity>,
+  ) {}
+
+  findAll(): Promise<GuildRequestEntity[]> {
+    return this.guildRequestRepository.find();
+  }
+
+  async findAllGuildRequests(userId: string): Promise<GuildRequestEntity[]> {
+    console.log(userId);
```
Comment (mrjvs) [inline vivid-server/src/modules/guildrequest/guildrequest.service.ts]: remove debug log

File: vivid-server/src/modules/guilds/guilds.controller.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  async seeAll() {
+    return this.guildsService.getAll();
+  }
+
+  @Post('change_anagram/:anagram')
+  async changeGuildAnagram(
+    @Param('anagram') anagram: string,
+    @User() user: UserEntity,
+  ): Promise<UserEntity>{
+    if (anagram.length > 5) throw new BadRequestException();
```
Comment (mrjvs) [inline vivid-server/src/modules/guilds/guilds.controller.ts]: make sure anagrams are only valid characters (use a regex like `/^[A-Z1-9\-]{1,5}$/g`)

Comment (mrjvs) [reply vivid-server/src/modules/guilds/guilds.controller.ts]: (cuz urls can also have stuff like null characters in the middle)

File: vivid-server/src/modules/users/user.service.ts [hunk 3]
```diff
@@ -16,13 +20,17 @@ export class UserService {
...
+  async findUser(id: string): Promise<UserEntity | void> {
+    return await this.userRepository
+      .findOne({
+        relations: ['joined_channels', 'joined_channels.channel'],
+        where: {
+          id,
+        },
+      })
+      .catch((error) => {
+        if (error.code === '22P02') throw new NotFoundException();
```
Comment (mrjvs) [inline vivid-server/src/modules/users/user.service.ts:35]: add default throw and remove void from return value

File: vivid-server/src/modules/guilds/guilds.service.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      .where('owner = :owner', { owner: userId })
+      .execute()
+      .catch((error) => {
+        if (error.code === '23505') throw new BadRequestException();
+        throw error;
+      });
+  }
+
+  async guildWin(guildAnagram: string, points: number): Promise<UpdateResult> {
+    console.log('winner: ', guildAnagram, points);
```
Comment (mrjvs) [inline vivid-server/src/modules/guilds/guilds.service.ts]: remove debug log

File: vivid-server/src/models/user.entity.ts [hunk 5]
```diff
@@ -24,7 +30,13 @@ export class UserEntity {
   @Column()
   intra_id: string;
 
+  @Column('varchar', { default: null })
+  guild_anagram: string;
```
Comment (mrjvs) [inline vivid-server/src/models/user.entity.ts]: dont save the anagram on the user, that can cause desyncs. instead use a relation (check joined_channels for example)

File: vivid-server/package.json [hunk 6]
```diff
@@ -25,6 +25,7 @@
     "class-transformer": "^0.4.0",
     "class-validator": "^0.13.1",
     "connect-typeorm": "^1.1.4",
+    "dto-mapping": "^1.1.0",
```
Comment (mrjvs) [inline vivid-server/package.json]: what do you use dto-mapping for?

**Review by mrjvs — CHANGES_REQUESTED** [review]

**Comment by mrjvs** [comment]
also make sure to fix the linting before merge

File: vivid-server/src/modules/users/user.service.ts [hunk 7]
```diff
@@ -16,13 +20,17 @@ export class UserService {
...
+      });
```
Comment (pmerku) [inline vivid-server/src/modules/users/user.service.ts]: Default throw to add, and remove void from return type

File: vivid-server/src/models/user-update.dto.ts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+export class UserUpdateDto {
+  name: string;
```
Comment (pmerku) [inline vivid-server/src/models/user-update.dto.ts]: You can add a decorator from `class-validator` package to constrain the string like:
```ts
@IsNotEmpty()
name: string;
```

**Review by pmerku — CHANGES_REQUESTED** [review]

File: vivid-server/src/modules/users/user.service.ts (hunk 7 above)
Comment (pmerku) [reply vivid-server/src/modules/users/user.service.ts]: Ah I see Jelle found it too

**Review by pmerku — APPROVED** [review]

**Review by mrjvs — APPROVED** [review]

**Pull request merged**